ui/src/pages/AllStories.tsx

```diff
@@ -21,111 +85,247 @@ function createData(
   is_visible: boolean,
   author_first_name: string,
   author_last_name: string,
-  author_country: string
-) {
+  author_country: string,
+  image_url: string,
+  video_url: string,
+  content: string
+): StoryView {
   const author_name = `${author_first_name} ${author_last_name}`;
   return {
-    id,
+    ID: id,
     title,
     current_city,
     year,
     author_name,
+    author_first_name,
+    author_last_name,
     author_country,
     is_visible,
+    image_url,
+    video_url,
+    content,
   };
 }
 
-const StyledAppBar = styled(AppBar)`
-  //styling
-`;
+interface State {
```

Comment:
kk let's extract this the reducer/action/state logic into another file plz. A folder structure like this?

```
pages
  ...
  AllStories -> folder
    index.tsx -> has the component
    reducer.ts -> has the reducer + state, action types
  ...
```

Reply:
You can also have a `types.tsx` file to protect against potenial circular dependencies.

Reply:
Also great time to write unit tests for the reducer (check out what I did here: https://github.com/uwblueprint/shoe-project/pull/280/files#diff-013eb0dfde8f991d00c2f99a2ab0539ab325daba7b2905cfb69f108c5b255979R7)

Reply:
ooh yes that cleans it up nicely

Reply:
thanks for the example, added some tests ✨  - wasn't too sure how to mock the ones requiring event listeners 
![image](https://user-images.githubusercontent.com/19617248/113665821-283b0600-967c-11eb-9939-840b4b9097a6.png)